clippy_lints/src/literal_string_with_formatting_args.rs

```diff
@@ -108,7 +108,22 @@ impl LateLintPass<'_> for LiteralStringWithFormattingArg {
                     if error.span.end >= current.len() {
                         break;
                     }
-                    current = &current[error.span.end + 1..];
+                    let mut pos = 0;
+                    // We find the closest char to where the error location ends.
+                    for (char_pos, _) in current.char_indices() {
```

Comment:
My first thought was to use [`ceil_char_boundary`](https://doc.rust-lang.org/std/primitive.str.html#method.ceil_char_boundary), also `is_char_boundary` exists. Any reason to not use them?

Reply:
I didn't know they existed. :o

Thanks!